pkg/controller/master/storagenetwork/storage_network.go

```diff
@@ -813,3 +820,56 @@ func (h *Handler) updateLonghornStorageNetwork(storageNetwork string) error {
 	}
 	return nil
 }
+
+func (h *Handler) cleanUpOldSNIPPool() error {
+	ipPools, err := h.whereaboutsCNIIPPoolCache.List(util.KubeSystemNamespace, labels.Everything())
+	if err != nil {
+		return err
+	}
+
+	nads, err := h.networkAttachmentDefinitions.List(util.HarvesterSystemNamespaceName, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	migrationPoolNames := h.getMigrationPoolNames(nads.Items)
+	for _, ipPool := range ipPools {
+		if _, keep := migrationPoolNames[ipPool.Name]; keep {
+			continue
+		}
+		err := h.whereaboutsCNIIPPoolClient.Delete(util.KubeSystemNamespace, ipPool.Name, &metav1.DeleteOptions{})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (h *Handler) getMigrationPoolNames(nads []nadv1.NetworkAttachmentDefinition) map[string]bool {
+	migrationPools := make(map[string]bool)
+	for _, nad := range nads {
+		annotations := nad.GetAnnotations()
+
+		if val, ok := annotations[kubevirt.VMMigrationNetworkPrefix]; !ok || val != "true" {
+			continue
+		}
+		var config network.BridgeConfig
+		if err := json.Unmarshal([]byte(nad.Spec.Config), &config); err != nil {
```

Comment:
log a message to indicate the error, as we skip the error

Reply:
logged error :heavy_check_mark: